nat-lab/tests/conftest.py

```diff
@@ -232,9 +233,75 @@ async def _copy_vm_binaries_if_needed(items):
                 return
 
 
+def save_dmesg_from_host(suffix):
+    try:
+        result = subprocess.run(
+            ["sudo", "dmesg", "-d", "-T"],
+            capture_output=True,
+            text=True,
+            check=True,
+        ).stdout
+    except subprocess.CalledProcessError as e:
+        print(f"Error executing dmesg: {e}")
+        return
+
+    if result:
+        with open(
+            os.path.join("logs", f"dmesg-{suffix}.txt"), "w", encoding="utf-8"
+        ) as f:
+            f.write(result)
+
+
+def save_audit_log(suffix):
+    try:
+        source_path = "/var/log/audit/audit.log"
+        if os.path.exists(source_path):
+            shutil.copy2(source_path, f"logs/audit_{suffix}.log")
+        else:
+            print(f"The audit file {source_path} does not exist.")
+    except Exception as e:  # pylint: disable=broad-exception-caught
+        print(f"An error occurred when processing audit log: {e}")
+
+
+async def _save_macos_logs(conn, suffix):
+    try:
+        dmesg_proc = await conn.create_process(["dmesg"]).execute()
+        with open(
+            os.path.join("logs", f"dmesg-macos-{suffix}.txt"), "w", encoding="utf-8"
```

Comment:
question: these logs are stored on the mac machine itself, but how do the same filename not clash with other log files from other mac machines for example when collecting those? If you could explain that part a bit that would be :heart:

Reply:
There is no other mac machine, only one macos test vm. `dmesg` command is executed from `linux vm` to `macos vm` via ssh, then stdout output is flushed to file in `linux vm`. During all pytest session it will create only 1 before and after file.

Reply:
Not sure if I got it right but in case there will appear a second mac machine, does it mean this code needs fixes then?

Reply:
maybe yes, but there is no plans for 2nd mac vm in any near or far future :smile: